### DISCUSSION TITLE
Gradle vs Maven?

### DISCUSSION THREAD

**Discussion by wolski0420** [M-Szymczyk/ProjektPortfel#26] [Q&A]
Rzecz jest taka, że te pliki gradlowskie o których wspominałeś @M-Szymczyk to chyba muszą zostać. Nie mogę znaleźć konkretnej informacji o tym w dokumentacji, ale niemal każde repo oparte na Gradle ma właśnie te pliki co my + przynosi to jakieś benefity. 

Zamiast rozwodzić się nad słusznością każdego pliczku gradlowskiego proponuję zmigrować na Mavena. 
1. Zredukujemy liczbę plików do minimum (na repo wystarczy sam pom.xml), w przeciwieństwie do Gradle
2. Maven traktuje każdą dependencję jako dependencję do projektu po prostu, w Gradlu zaś każda dependencja jest podzielona na jakieś podgrupy implementation, testImplementation i testRuntimeOnly. To może i coś mówi ale czy to potrzebne? Nie wspomnę już o javaFx która jest oddzielnym bytem w tym frameworku. Maven to po prostu ma <dependencyManagement> lub <dependencies> i elo
3. Maven szybciej się buduje z tego co widzę lokalnie u siebie.
4. Jedyna wada Mavena względem Gradle jest taka, że build.gradle z reguły jest mniejszy niż pom.xml. Ale przecież to plik konfiguracyjny a nie klasa Javowa...

Co o tym sądzisz? Dużo z tym roboty nie powinno być, ostatnio siedzę sporo w Mavenie więcej więc raczej wiem co trzeba zrobić

**Comment by M-Szymczyk** [comment]
Zamiast czytać odpowiedzi na pytania to jeżeli obaj wiemy jak korzystać z Mavena, a z Gradle już nie koniecznie ,to trzeba przejść na Mavena. 

_Rzecz jest taka, że te pliki gradlowskie o których wspominałeś @M-Szymczyk to chyba muszą zostać._ <- Jakie pliki gradlowskie? W sensie cały katalog _gradle/wrapper, build.gradle, gradlew, gradlew.bat_? 

_Zredukujemy liczbę plików do minimum (na repo wystarczy sam pom.xml), w przeciwieństwie do Gradle_ <- popieram, ale w _pom.xml_ przydaje się komentowanie sekcji co do czego jest. Potem może się zdarzyć, że tam jest za dużo zależności i zamiast po kolei sprawdzać co jest do usunięcia to tylko usuwasz całe sekcje jeżeli z tego nie korzystasz.

 _Maven traktuje każdą dependencję jako dependencję do projektu po prostu, w Gradlu zaś każda dependencja jest podzielona na jakieś podgrupy implementation, testImplementation i testRuntimeOnly. To może i coś mówi ale czy to potrzebne?_ <- Jeżeli i tak nie wiemy do czego to wykorzystać to lepiej od razu usunąć i mieć problem z głowy.
